components/org.wso2.bfsi.consent.management.common/src/main/java/org/wso2/bfsi/consent/management/common/caching/ConsentManagementBaseCache.java

```diff
@@ -0,0 +1,212 @@
+/**
+ * Copyright (c) 2024, WSO2 LLC. (https://www.wso2.com).
+ * <p>
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.bfsi.consent.management.common.caching;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.bfsi.consent.management.common.exceptions.ConsentManagementException;
+
+import java.util.concurrent.TimeUnit;
+
+import javax.cache.Cache;
+import javax.cache.CacheBuilder;
+import javax.cache.CacheConfiguration;
+import javax.cache.CacheManager;
+import javax.cache.Caching;
+
+/**
+ * Abstract cache manager for Consent Management.
+ *
+ * @param <K>   Extended Cache Key
+ * @param <V>   Cache Value
+ */
+public abstract class ConsentManagementBaseCache<K extends ConsentManagementBaseCacheKey, V> {
+
+    private static final String BASE_CACHE_KEY = "CONSENT_MGT_BASE_CACHE";
+    private final String cacheName;
+
+    private static final Log log = LogFactory.getLog(ConsentManagementBaseCache.class);
+
+    /**
+     * On Demand Retriever for caching.
+     */
+    public interface OnDemandRetriever {
+
+        Object get() throws ConsentManagementException;
+    }
+
+    /**
+     * Initialize With unique cache name.
+     *
+     * @param cacheName   unique cache name.
+     */
+    public ConsentManagementBaseCache(String cacheName) {
+
+        this.cacheName = cacheName;
+
+        if (log.isDebugEnabled()) {
+            log.debug(String.format("Base Cache initialized for %s", cacheName.replaceAll("[\r\n]", "")));
+        }
+    }
+
+    /**
+     * Get from cache or invoke ondemand retriever and store.
+     *
+     * @param key                  cache key.
+     * @param onDemandRetriever    on demand retriever.
+     * @return                   cached object.
+     * @throws ConsentManagementException  if an error occurs while retrieving the object
+     */
+    public V getFromCacheOrRetrieve(K key, OnDemandRetriever onDemandRetriever) throws ConsentManagementException {
+
+        Cache<K, V> cache = getBaseCache();
+
+        if (cache.containsKey(key)) {
+
+            if (log.isDebugEnabled()) {
+                log.debug(String.format("Found cache entry `%s` in cache %s",
+                        key.toString().replaceAll("[\r\n]", ""), cacheName.replaceAll("[\r\n]", "")));
+            }
+            return cache.get(key);
+        } else {
```

Comment:
Can't we replace this with `getFromCache' method?

Reply:
Fixed in https://github.com/wso2-extensions/bfsi-consent-management/pull/5/commits/4d9138c332549a29ab817615c49fd31c6d8cc121